.github/workflows/ci.yml

```
@@ -2,39 +1,161 @@
+
+    - name: Upload security report
+      uses: actions/upload-artifact@v3
+      with:
+        name: security-report
+        path: bandit-report.json
+
+  build:
+    needs: [test, security]
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)